### DISCUSSION TITLE
Update disarm code with something like keymaster?

### DISCUSSION THREAD

**Discussion by MuscleGeek69** [tsightler/ring-mqtt#617] [General Discussion]
Hello guys,

Im planning to use keymaster with AirBnB and Vrbo caledars to create the guest code based on the last four of the phone number  in the reservation and I was wondering if I can also update the disarm code inside the Ring-MQTT since I have a wall panel that people can use to disarm and arm the house. 

Thanks in advance.

**Comment by tsightler** [comment]
It's not currently possible with ring-mqtt as, currently, ring-client-api library does not have any functions for this.  It could be figured out, but it's just not an area I've tried to reverse engineer much.

**Comment by MuscleGeek69** [comment]
Thank you for your reply.

That will be even better if we can create user code from HA using keymaster. I was actually looking to just update the disarm code in HA, the one the Ring-Mqtt supports. I am open to use AppDeamon if it is the only solution.

Thanks,

On Apr 26, 2023, at 8:16 AM, tsightler ***@***.***> wrote:

﻿

It's not currently possible with ring-mqtt as, currently, ring-client-api library does not have any functions for this. It could be figured out, but it's just not an area I've tried to reverse engineer much.

**Reply by tsightler** [reply]
The code in HA is just sent as part of MQTT discovery.  You could update it by sending your own MQTT discovery message with a new code.  If you wanted to do it with REST API you could use the HA MQTT publish service, which has an API, to send an MQTT message.
